### PR TITLE
Add the resource type on RelatedResourcesPanel

### PR DESCRIPTION
**Pull request by GretaD**
### ☑️ Resolves

* Fix #https://github.com/nextcloud/related_resources/pull/303

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A

### 🚧 Tasks

- [ ] ...

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Comment by Pytal** [comment]
Also needs the new `limit` query param from https://github.com/nextcloud/related_resources/pull/303 @GretaD

File: src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue [hunk 1]
```diff
@@ -147,18 +158,24 @@ export default {
...
+			let limit = null
 
 			if (this.isFiles) {
 				providerId = 'files'
 				itemId = this.fileInfo.id
 			} else {
 				providerId = this.providerId
 				itemId = this.itemId
+				resourceType = this.resourceType
+				limit = this.limit
```
Comment (Pytal) [inline src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue:171]:
```suggestion
```

File: src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue [hunk 2]
```diff
@@ -147,18 +158,24 @@ export default {
...
 			}
 
-			return generateOcsUrl('/apps/related_resources/related/{providerId}?itemId={itemId}&format=json', {
+			return generateOcsUrl('/apps/related_resources/related/{providerId}?itemId={itemId}&resourceType={resourceType}&limit={limit}&format=json', {
 				providerId,
 				itemId,
+				resourceType,
+				limit,
```
Comment (Pytal) [inline src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue:178]:
```suggestion
				resourceType: this.resourceType,
				limit: this.limit,
```

File: src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue [hunk 3]
```diff
@@ -89,6 +89,17 @@ export default {
...
 			default: null,
 		},
+		/**
+		 * Limits to specific resource type. i.e. any provider id implemented with `\OCA\RelatedResources\IRelatedResourceProvider::getProviderId()`
+		 */
+		resourceType: {
+			type: String,
+			default: null,
+		},
+		limit: {
```
[the remark this answers is not in the document]
Comment (GretaD) [reply src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue:99]: I was not sure what to write :) but i'll think something

File: src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue [hunk 4]
```diff
@@ -89,6 +89,20 @@ export default {
...
 		},
+		/**
+		 * Limits to specific resource type. i.e. any provider id implemented with `\OCA\RelatedResources\IRelatedResourceProvider::getProviderId()`
+		 */
+		resourceType: {
+			type: String,
+			default: null,
+		},
+		/**
+		 * Changes the number of results for the resourceType
```
Comment (Pytal) [inline src/components/NcRelatedResourcesPanel/NcRelatedResourcesPanel.vue:100]:
```suggestion
		 * Set the maximum number of resources to load
```
Limit does not only apply when resourceType is set

**Review by Pytal — APPROVED** [review]
Other than minor remark

**Pull request merged**